web-api/src/persistence/postgres/utils/mutex.ts

```diff
@@ -1,146 +1,100 @@
-import { ALLOWLIST_FEATURE_FLAGS } from '../../../../shared/src/business/entities/EntityConstants';
+import crypto from 'crypto';
 import { ServerApplicationContext } from '@web-api/applicationContext';
-import { ServiceUnavailableError } from '@web-api/errors/errors';
 import { UnknownAuthUser } from '@shared/business/entities/authUser/AuthUser';
 import { getLogger } from '@web-api/utilities/logger/getLogger';
+import { tryGetLock } from '@web-api/persistence/postgres/utils/operation/tryGetLock';
+import { releaseLock } from '@web-api/persistence/postgres/utils/operation/releaseLock';
+import { ServiceUnavailableError } from '@web-api/errors/errors';
 import { sleep } from '@shared/tools/helpers';
+import { getScopedDbConnection } from '@web-api/getConnection';
+import { settlePromises } from '@web-api/utilities/settlePromises';
 
-export const checkLock = async ({
-  applicationContext,
-  identifier,
-}: {
-  applicationContext: ServerApplicationContext;
-  identifier: string;
-}): Promise<boolean> => {
-  const featureFlags = await applicationContext
-    .getUseCases()
-    .getAllFeatureFlagsInteractor(applicationContext);
-
-  const isCaseLockingEnabled =
-    featureFlags[ALLOWLIST_FEATURE_FLAGS.ENTITY_LOCKING_FEATURE_FLAG.key];
-
-  const currentLock = await applicationContext
-    .getPersistenceGateway()
-    .getLock({ applicationContext, identifier });
-
-  if (!currentLock) {
-    getLogger().warn('Entity is NOT currently locked', { identifier });
-    return false;
-  }
-
-  getLogger().warn('Entity is currently locked', { currentLock });
-
-  if (!isCaseLockingEnabled) {
-    return false;
-  }
-
-  return true;
+/**
+ * Converts a string into a consistent 32-bit integer to use as an advisory lock ID.
+ */
+export const hashLockId = (input: string): number => {
```

Comment:
Debugging this is going to be very, very hard. I wonder if it is worth avoiding a hash and making an invertible map so that we could go from some inscrutable integer to a meaningful string like `case|101-20`. E.g. (probably a terrible idea off the top of my head) represent every letter (case-insensitive) as a two digit code (a = 11, b=12...) with 0 as a separator. Something like that would preserve uniqueness but also allow us, if needed, to know what was actually locked.

Reply:
That is a good idea. The only thing that stopped my from putting effort into something like that in the first place, is that the locks are session-scoped, meaning that they only live for the duration of the time that `acquireLock` is live.  Still, you're right, hashing them makes them unnecessarily inscrutable

Reply:
Or if we hash, we could save the original identifiers in a lock object.

Reply:
Curious about that last idea. Save them in the DB? We do keep them in the `identifierObjects` for logging

Reply:
🤦